cmd/serve/serve.go

```diff
@@ -101,12 +102,14 @@ func New(log *logrus.Entry) *cobra.Command {
 				int(eventQueueSize),
 				log,
 			)
+			ctx, cancel := context.WithCancel(context.Background())
+			setupSigHandler(ctx, cancel)
```

Comment:
i don't think cancel is called in all cases with this setup (crashing or other error without ctrl-c)